north/src/runtime/process.rs

```diff
@@ -433,64 +455,27 @@ pub mod minijail {
         // with MS_NODEV | MS_NOEXEC | MS_NOSUID
         jail.remount_proc_readonly();
 
-        let mounts = if let Some(ref mounts) = container.manifest.mounts {
-            mounts
-        } else {
-            return Ok(());
-        };
-
-        for mount in mounts {
-            match mount.r#type {
-                // Bind mounts
-                MountType::Bind => {
-                    if let Some(ref source) = mount.source {
-                        // Check if the source exists. If not issue a warning
-                        if source.exists() {
-                            let source: PathBuf = source.clone().into();
-                            let target: PathBuf = mount.target.clone().into();
-                            let rw = mount
-                                .flags
-                                .as_ref()
-                                .map(|flags| flags.contains(&MountFlag::Rw))
-                                .unwrap_or_default();
-                            mount_bind(jail, &source, &target, rw)?;
-                        } else {
-                            warn!(
-                                "Cannot bind mount nonexitent source {} to {}",
-                                source.display(),
-                                mount.target.display()
-                            );
-                        }
-                    } else {
-                        // TODO: Bind mounts without a source are invalid and should be checked in Manifest::verify
-                        return Err(anyhow!("Cannot mount of type bind without source"));
+        for (target, mount) in container.manifest.mounts() {
+            let source: PathBuf = match mount.host.as_ref() {
+                Some(dir) => {
+                    if !dir.exists() {
+                        warn!("Creating nonexistent input directory {}", dir.display(),);
```

Comment:
Think this is a error. If the mounts distinguish between `Persist` and `BindMount` it is definitely an error.